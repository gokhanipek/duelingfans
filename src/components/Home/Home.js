import React from 'react'
import { Link } from 'react-router-dom';
import './Home.scss';
import curry from './../../assets/curry.png';

const Home = () => {
    return (
        <div className="home__page--wrapper">
            <div className="home__page--banner">
                <h2 className="home__page--title">One way to <span className="home__page--colored">win</span></h2>
                <img className="home__page--player" src={curry} alt="cover" />
            </div>
            <Link className="home__page--button" to="/duel">Play Now!</Link>
        </div>
    )
}

export default Home
