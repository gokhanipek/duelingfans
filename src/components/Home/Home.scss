@import url('//db.onlinewebfonts.com/c/557661ea24d10fa4d2d85f60f04411d8?family=Shentox');
.home__page--wrapper {
    display: flex;
    width: 100%;
    height: 100vh;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    @media screen and (max-width: 992px) {
        height: 70vh;
    }
    @media screen and (max-width: 375px) {
        height: 90vh;
    }
    .home__page--banner {
        display: flex;
        align-self: center;
        @media screen and (max-width: 992px) {
            flex-direction: column;
            justify-content: center;
            align-items: center;
        }
        .home__page--title {
            max-width: 3.15em;
            margin: 0;
            font-size: 10vh;
            line-height: .85;
            text-align: right;
            position: relative;
            font-family: "Shentox";
            color: #fff;
            @media screen and (max-width: 992px) {
                font-size: 7vh;
                text-align: center;
            }
            .home__page--colored {
                margin-top: -0.025em;
                margin-bottom: -0.25em;
                display: block;
                color: #3de585;
                font-size: 1em;
                background: -webkit-linear-gradient(270deg, #81edaf 13%, #33e37e 44%, #1e9448 90%);
                -webkit-background-clip: text;
                -webkit-text-fill-color: transparent;
                position: relative;
                text-shadow: none;
                &:before {
                    content: "win";
                    position: absolute;
                    z-index: -1;
                    text-shadow: -0.025em -0.025em 4px rgba(0, 0, 0, .25);
                }
            }
        }
        .home__page--player {
            height: 225px;
            @media screen and (max-width: 992px) {
                width: 200px;
                height: auto;
            }
        }
    }
    .home__page--button {
        margin: 0;
        padding: 1.5em 0;
        width: 45%;
        margin-top: 20px;
        max-width: 300px;
        border-radius: 6px;
        border: 2px solid #1493ff;
        background: #1f375b;
        font-family: "Shentox";
        font-size: .75rem;
        line-height: 1;
        text-align: center;
        text-transform: uppercase;
        box-shadow: 0 4px 6px rgba(0, 0, 0, .4);
        -webkit-transition: all .2s ease;
        text-decoration: none;
        color: #fff;
        transition: all .2s ease;
        position: relative;
        z-index: 1;
        &:after {
            content: "";
            position: absolute;
            z-index: -1;
            top: -2px;
            right: -2px;
            bottom: -2px;
            left: -2px;
            opacity: 0;
            border-radius: 6px;
            box-shadow: 0 12px 16px 4px rgba(0, 0, 0, .2);
            -webkit-transition: all .2s ease;
            transition: all .2s ease;
        }
        &:hover {
            transform: scale(1.01);
            color: #30e37b;
        }
    }
    .home__page--info {
        width: 60%;
        color: #fff;
        span {
            display: block;
            font-size: 0.4em;
        }
    }
}