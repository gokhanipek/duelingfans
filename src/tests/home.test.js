import React from 'react';
import { shallow } from 'enzyme';
import Home from './../components/Home/Home';

describe('Home component', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Home debug />);
  
    expect(component).toMatchSnapshot();
  });
});
