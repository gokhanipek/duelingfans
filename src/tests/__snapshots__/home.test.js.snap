// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render correctly in "debug" mode 1`] = `
<div
  className="home__page--wrapper"
>
  <div
    className="home__page--banner"
  >
    <h2
      className="home__page--title"
    >
      One way to 
      <span
        className="home__page--colored"
      >
        win
      </span>
    </h2>
    <img
      alt="cover"
      className="home__page--player"
      src="curry.png"
    />
  </div>
  <Link
    className="home__page--button"
    to="/duel"
  >
    Play Now!
  </Link>
</div>
`;
